packages/studio-base/src/panels/ThreeDeeRender/transforms/TransformTree.ts

```
@@ -144,12 +152,18 @@ export class TransformTree {
     }
   }
 
-  public hasFrame(id: string): boolean {
+  public hasFrame(id: CoordinateFrame<AnyFrameId>["id"]): boolean {
+    if (id === CoordinateFrame.FALLBACK_FRAME_ID) {
+      return true;
```

I wonder if we should actually return false if there are some user frames present (and return undefined from `frame()`)? Not sure it would make a difference in practice though.

---

Hmmm. I think I'm going to leave it as is since if we're ever checking for a fallback frame's existence in the transform tree (which we shouldn't be), we will probably try to find another fallback frame or early return if this returns false which might hide the fact that we're checking it. I think showing something in the fallback frame will be a bit more obvious of an error because none of the other transforms will have a path to it in the tree